### PR TITLE
New package: v2ray-domain-list-community-20220809072217

### PR DESCRIPTION
**Pull request by ketlrznt**
#### Testing the changes
- I tested the changes in this PR: **YES**

#### New package
- This new package conforms to the [quality requirements](https://github.com/void-linux/void-packages/blob/master/Manual.md#quality-requirements): **YES**

#### Local build testing
- I built this PR locally for my native architecture, (x86_64-glibc)
- I built this PR locally for these architectures
  - i686 (native masterdir)
  - i686 (crossbuild)

### CODE REVIEW

**Comment by ketlrznt** [comment]
Notes: 
- This package adds additional geosites for routing purposes in [New package: v2ray-4.44.0 #31653](https://github.com/void-linux/void-packages/pull/31653), which are also needed by [ New package: v2rayA-1.5.9.1698.1 #38528 ](https://github.com/void-linux/void-packages/pull/38528).
- I haven't test for crossbuilds because [template #16](https://github.com/ketlrznt/void-packages/blob/027c40ed6dc6031d048ad1d4f6af11f855a6280c/srcpkgs/v2ray-domain-list-community/template#L16) includes path which doesn't match `crossbuild` folder path on non-native machine. I tested build in `i686 native masterdir`, seems everything goes fine.
- After solving [v2ray build failure](https://github.com/void-linux/void-packages/pull/38538#issuecomment-1209414575), I think this package can add to `v2ray`'s `depends` list.

**Comment by classabbyamp** [comment]
use something like this to fix the 2nd bullet point

https://github.com/void-linux/void-packages/blob/a15f0148180c7cb23809080cc14ccdade54e0087/srcpkgs/kubernetes-kind/template#L18

**Comment by ketlrznt** [comment]
> use something like this to fix the 2nd bullet point
> 
> https://github.com/void-linux/void-packages/blob/a15f0148180c7cb23809080cc14ccdade54e0087/srcpkgs/kubernetes-kind/template#L18

Done.

File: srcpkgs/v2ray-domain-list-community/template [hunk 1]
```diff
@@ -0,0 +1,21 @@
...
+short_desc="Community managed domain list for Project V"
+maintainer="ketlrznt <tansuanyinliao8888@gmail.com>"
+license="MIT"
+homepage="https://github.com/v2fly/domain-list-community"
+distfiles="https://github.com/v2fly/domain-list-community/archive/refs/tags/${version}.tar.gz"
+checksum=3583a9dd91ee9d2f514e5ee49360c579ae52d2a3b1f4693ebd6bc0e819cabfe0
+
+post_install() {
+	local binary_file_path=$(find ${GOPATH}/bin/ -name domain-list-community)
+	$binary_file_path --datapath=${wrksrc}/data --outputname geosite.dat
```
Comment (ketlrznt) [inline srcpkgs/v2ray-domain-list-community/template:17]: I think this line leads to build failures in CI. xbps-src builds binary file successfully, but what I want is the `dat` file generates by go build binary output.

Comment (ketlrznt) [reply srcpkgs/v2ray-domain-list-community/template:17]: A method to solve this is fetching `dat` file released by upstream directly,  But in this way the build process doesn't make sense.

Comment (classabbyamp) [reply srcpkgs/v2ray-domain-list-community/template:17]: 1. I think what's happening is you're trying to run a binary built for not-x86_64 on x86_64, which will only work in a few cases.
2. yes, I think fetching the released dat file is better, and can probably be simply added as a distfile for v2ray (but I'm not sure, I don't know anything about this software)

Comment (ketlrznt) [reply srcpkgs/v2ray-domain-list-community/template:17]: > 1. I think what's happening is you're trying to run a binary built for not-x86_64 on x86_64, which will only work in a few cases.
> 
>     2. yes, I think fetching the released dat file is better, and can probably be simply added as a distfile for v2ray (but I'm not sure, I don't know anything about this software)

Done. Thank you for your advice.

For list point 2, `v2ray-domain-list-community` gets updates very rapidly, as upstream releases new dat file every week. It's better to make a separated package for it, or v2ray may gets revision very often, which doesn't mean too much.

Comment (classabbyamp) [reply srcpkgs/v2ray-domain-list-community/template:17]: does the list need to be updated every week? either way you're going to be doing updates very often if so

Comment (ketlrznt) [reply srcpkgs/v2ray-domain-list-community/template:17]: > does the list need to be updated every week? either way you're going to be doing updates very often if so


I'd like to take this responsibility. But frankly speaking, a somewhat outdated list can also provides functionalities. I will update this template as soon as possible after new upstream release.

Comment (classabbyamp) [reply srcpkgs/v2ray-domain-list-community/template:17]: > I'd like to take this responsibility

do remember that someone from the maintainer team would have to review and merge the request each week

Comment (ketlrznt) [reply srcpkgs/v2ray-domain-list-community/template:17]: > > I'd like to take this responsibility
> 
> do remember that someone from the maintainer team would have to review and merge the request each week
I will. Another way is to make the update cycle parallel to v2ray (if possible), I will do some works to test this file and make everything works fine. May not disturb maintainers very often then.

**Comment by ketlrznt** [comment]
I forgot to fetch LICENSE and install it. I will update this commit later.

**Comment by classabbyamp** [comment]
this is a bit cleaner

```bash
# Template file for 'v2ray-domain-list-community'
pkgname=v2ray-domain-list-community
version=20220809072217
revision=1
build_style=fetch
short_desc="Community managed domain list for Project V"
maintainer="ketlrznt <tansuanyinliao8888@gmail.com>"
license="MIT"
homepage="https://github.com/v2fly/domain-list-community"
distfiles="https://github.com/v2fly/domain-list-community/releases/download/${version}/dlc.dat
 https://raw.githubusercontent.com/v2fly/domain-list-community/master/LICENSE"
checksum="fb7312ba36be5865b14858f44dff3dadf996eb5623bd23d63ead448feb40622d
 b9d84a22870d3f21c91a4c6e410c9cc51d00902f5233ad0c84011479244bf7d2"

do_install() {
	vinstall dlc.dat 0644 usr/share/v2ray geosite.dat
	vlicense LICENSE
}
```

**Comment by ketlrznt** [comment]
> this is a bit cleaner
> 
> ```shell
> # Template file for 'v2ray-domain-list-community'
> pkgname=v2ray-domain-list-community
> version=20220809072217
> revision=1
> build_style=fetch
> short_desc="Community managed domain list for Project V"
> maintainer="ketlrznt <tansuanyinliao8888@gmail.com>"
> license="MIT"
> homepage="https://github.com/v2fly/domain-list-community"
> distfiles="https://github.com/v2fly/domain-list-community/releases/download/${version}/dlc.dat
>  https://raw.githubusercontent.com/v2fly/domain-list-community/master/LICENSE"
> checksum="fb7312ba36be5865b14858f44dff3dadf996eb5623bd23d63ead448feb40622d
>  b9d84a22870d3f21c91a4c6e410c9cc51d00902f5233ad0c84011479244bf7d2"
> 
> do_install() {
> 	vinstall dlc.dat 0644 usr/share/v2ray geosite.dat
> 	vlicense LICENSE
> }
> ```

Thank you! As a packaging newbie I referred Manual and `xnew` to write template, section [Optional variables](https://github.com/void-linux/void-packages/blob/master/Manual.md#optional-variables) indicates that `distfile` should be compressed data. I don't know such usage. Now I know it.

**Pull request closed without merging**